services/alerts/src/AlertsManagerContext.ts

```diff
@@ -0,0 +1,62 @@
+import React from 'react'
+import { Alert, AlertsManager, AlertsManagerAlert } from './types'
+
+type SetAlertsFunction = React.Dispatch<React.SetStateAction<Alert[]>>
+
+export const makeAlertManager = (
+    setAlerts: SetAlertsFunction
+): AlertsManager => {
+    let id = 0
+
+    const remove = (alert: Alert) => {
+        setAlerts(alerts => {
+            const idx = alerts.findIndex(a => a === alert)
+
+            if (idx === -1) {
+                return alerts
+            }
+
+            return [
+                ...alerts.slice(0, idx),
+                ...alerts.slice(idx + 1, alerts.length),
+            ]
+        })
+    }
+    const add = (alert: Alert) => {
+        setAlerts(alerts => {
+            if (alerts.some(a => a === alert)) {
+                return alerts
+            }
```

Comment:
Is the idea behind this to prevent alerts from being shown twice?
Then I'd prefer not to do this check at all.

If I click a UI element and it causes an error alert to show up and and click the same UI element again, I'd expect the UI to respond to that. Having the same error twice is duplicate information in regards to the content, but it allows the user to get a visual response when interacting with the app, which is different information that is important as well. 

IMO the app should take care of whether alerts can be shown twice or not.

Reply:
We could keep this functionality but put it behind a flag in the options

Reply:
I haven't verified this in the example app yet, but I think we can't do without this check. I think this is needed to prevent duplicate alerts from being displayed due to re-renders of the component that calls `show`. 

Also, when I started work on this service, I made a little helper to show alerts. It was just a text input with a button and when clicking the button it would show an alert with the text in the input. Clicking the button several times would produce a new alert every time. I'm mentioning this because it seems to me that the situation you are worried about is not actually a problem in the current implementation.

I'll look into it, but I'm fairly certain we need to keep this in.

Reply:
> I think this is needed to prevent duplicate alerts from being displayed due to re-renders of the component that calls show.

I would classify that as a bug in the app, just like as if `refetch` was called on every render. Therefore this check is unnecessary

Reply:
I was just staring at the current implementation and it's actually impossible that in its current form this is doing anything. We don't add the alert we receive directly to the alerts array, but we add a new object that is a constructed from the received alert plus some new properties. As such the condition `alerts.some(a => a === alert)` is guaranteed to return false, because this is comparing the alerts by reference and they are always different.

I am going to remove this check for now. If it turns out that we do need this functionality, then I think we should implement something based on shallow/deep object equality instead.